packages/eas-cli/src/branch/queries.ts

```diff
@@ -89,21 +97,16 @@ function renderPageOfBranches(
     });
 
     table.push(
-      ...currentPage.map(branch => {
-        const update: UpdateBranchFragment['updates'][0] | undefined = branch.updates[0];
-        return [
-          branch.name,
-          formatUpdate(update),
-          update?.runtimeVersion ?? 'N/A',
-          update?.group ?? 'N/A',
-          update?.group
-            ? getPlatformsForGroup({
-                updates: branch.updates,
-                group: update.group,
-              })
-            : 'N/A',
-        ];
-      })
+      ...currentPage.map(branch => [
+        branch.name,
+        formatUpdateMessage(branch.updates[0]),
+        branch.updates[0]?.runtimeVersion ?? 'N/A',
+        branch.updates[0]?.group ?? 'N/A',
+        getPlatformsForGroup({
+          group: branch.updates[0]?.group,
+          updates: branch.updates,
+        }),
+      ])
```

Comment:
Just confirming, this is a paginated table of branches that shows the branch name and info about most recent published update on the branch? I guess using `branch:view` is the way to see rest of the updates on the branch.

Also I'm assuming we have a limit how many updates we query per branch so technically the platforms column could be incorrect, but I don't think this will happen in most cases.

Either way, I think this is a lot cleaner if we hoist all the undefined handling out of the functions:

```suggestion
      ...currentPage.map(branch => {
        if (branch.updates.length === 0) {
          return [branch.name, 'N/A', 'N/A', 'N/A', 'N/A'];
        }

        const latestUpdateOnBranch = branch.updates[0];
        const latestUpdateGroup = latestUpdateOnBranch.group;
        const allUpdatesInGroup = branch.updates.filter(u => u.group === latestUpdateGroup);
        return [
          branch.name,
          formatUpdateMessage(latestUpdateOnBranch),
          latestUpdateOnBranch.runtimeVersion,
          latestUpdateOnBranch.group,
          getPlatformsForGroup(allUpdatesInGroup),
        ];
      })
```

Reply:
> Just confirming, this is a paginated table of branches that shows the branch name and info about most recent published update on the branch? I guess using branch:view is the way to see rest of the updates on the branch.

You got it, man.

> Also I'm assuming we have a limit how many updates we query per branch so technically the platforms column could be incorrect, but I don't think this will happen in most cases.

Yeah, this is a good point to consider. In `graphql/types/UpdateBranch.ts` we define `UpdateBranchFragment` getting the branch name, id, and the first 10 updates on each branch. So the platforms column will be correct in all cases. Additionally, `UpdateBranchFragment`  is only used in this query (`listBranchesAsync`/`listAndRenderBranchesOnAppAsync`).

Taking a closer look here, I think we should update the fragment to grab the first update group, instead of raw updates. There shouldn't really be breaking changes anyway, since it's only used in the `branches:list`.